test-d/extract-rest-element.ts

```diff
@@ -0,0 +1,58 @@
+import {expectType} from 'tsd';
+import type {ExtractRestElement} from '../index.d.ts';
+
+// Simple Rest element detection
+expectType<ExtractRestElement<['start', ...string[]]>>({} as string);
+expectType<ExtractRestElement<['foo', ...number[], true]>>({} as number);
+expectType<ExtractRestElement<['foo', ...boolean[], 'bar']>>({} as boolean);
+expectType<ExtractRestElement<[...Array<'foo'>, unknown]>>({} as 'foo');
+expectType<ExtractRestElement<['bar', ...Array<5>]>>({} as 5);
+
+// Leading Rest element
+expectType<ExtractRestElement<[...string[], 1]>>({} as string);
+expectType<ExtractRestElement<[...number[], 'a', 'b']>>({} as number);
+expectType<ExtractRestElement<[...any[], true]>>({} as any);
+expectType<ExtractRestElement<[...[], 'last']>>({} as never);
+expectType<ExtractRestElement<[...never[], 'end']>>({} as never);
+expectType<ExtractRestElement<[...unknown[], 2, 3]>>({} as unknown);
+
+// Middle Rest element
+expectType<ExtractRestElement<['a', ...string[], 'z']>>({} as string);
+expectType<ExtractRestElement<['x', ...boolean[], true]>>({} as boolean);
+expectType<ExtractRestElement<['x', ...any[], 'y']>>({} as any);
+
+// Trailing Rest element
+expectType<ExtractRestElement<[1, 2, ...string[]]>>({} as string);
+expectType<ExtractRestElement<['foo', ...Array<'bar'>]>>({} as 'bar');
+expectType<ExtractRestElement<[number, ...number[]]>>({} as number);
+
+// Rest element Only
```

Comment:
Try keeping consistent casing with comments as well, looks bad IMO.
```suggestion
// Rest element only
```

Reply:
I will one by one.